openlibrary/components/BulkSearch/utils/classes.js

```diff
@@ -37,15 +106,14 @@ export class BulkSearchState{
     constructor(){
         /** @type {string} */
         this.inputText= '';
-        /** @type {ExtractedBook[]} */
-        this.extractedBooks = [];
         /** @type {BookMatch[]} */
         this.matchedBooks = [];
         /** @type {MatchOptions} */
         this.matchOptions =  new MatchOptions()
-        /** @type {ExtractedOptions} */
+        /** @type {ExtractionOptions} */
         this.extractionOptions = new ExtractionOptions();
         this.errorMessage = []
+        this.extractionMethod = new AbstractExtractor()
```

Comment:
This should be null, and give a type hint that its type is `AbstractExtractor`. Most languages will actually prevent you from constructing an abstract class!
```suggestion
        this.extractionMethod = null
```

Reply:
Great fix, thank you.